### PR TITLE
Gdr 2442

### PR DESCRIPTION
**Pull request by bczech**
# Description
## What changed?
Related JIRA issue: GDR-2442

## Why was it changed?
To properly swap the data during converting MAE into raw data

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: R/convert_se_to_raw_data.R [hunk 1]
```diff
@@ -20,14 +20,34 @@ convert_mae_to_raw_data <- function(mae) {
   
   data <- gDRutils::MAEpply(mae, convert_se_to_raw_data)
   
+  if ("swap_sa" %in% names(merged_df)) {
```
Comment (gladkia) [inline R/convert_se_to_raw_data.R:23]: Do we have a unit test for this case?

Comment (bczech) [reply R/convert_se_to_raw_data.R]: Sorry, this chunk of code has been remove locally, I updated the PR

File: R/convert_se_to_raw_data.R [hunk 2]
```diff
@@ -23,13 +23,11 @@ convert_mae_to_raw_data <- function(mae) {
   # Remove duplicates shared between assays to keep only original single-agent
   common_records <- Reduce(intersect, lapply(data, "[[", "record_id"))
   sa_name <- gDRutils::get_experiment_groups("single-agent")[["single-agent"]]
-  data[setdiff(names(data), sa_name)] <- 
-    lapply(data[setdiff(names(data), sa_name)], function(x) {
-      x[!record_id %in% common_records]
-    })
-  
+  if (length(names(data) > 1)) {
```
Comment (j-smola) [inline R/convert_se_to_raw_data.R]: Please, add tests for scenario `length(names(data) > 1`

**Review by gladkia — APPROVED** [review]

**Review by gladkia — APPROVED** [review]

File: R/convert_se_to_raw_data.R (hunk 2 above)
Comment (bczech) [reply R/convert_se_to_raw_data.R]: Added in [2317075](https://github.com/gdrplatform/gDRcore/pull/145/commits/23170753f42755e1b865c6b0e1dcfd084f71b8bb)

File: tests/testthat/test-convert_se_to_raw_data.R [hunk 3]
```diff
@@ -28,6 +28,19 @@ test_that("convert_mae_to_raw_data works as expected with matrix data", {
...
   test_synthetic_data(mae$result, mae2$result, data)
+  
+  trt <- BumpyMatrix::unsplitAsDataFrame(SummarizedExperiment::assay(original[[1]], "RawTreated"))
+  trt <- trt[!duplicated(trt$record_id), ]
+  df_raw1 <- convert_mae_to_raw_data(original)
+  SummarizedExperiment::assay(original[[1]], "RawTreated") <- BumpyMatrix::splitAsBumpyMatrix(
+    trt[!colnames(trt) %in% c("row", "column")],
+    row = trt$row,
+    column = trt$column)
+  df_raw2 <- convert_mae_to_raw_data(original)
```
Comment (j-smola) [inline tests/testthat/test-convert_se_to_raw_data.R:43]: Why single-agent is completely removed here?

**Review by j-smola — APPROVED** [review]

File: tests/testthat/test-convert_se_to_raw_data.R [hunk 4]
```diff
@@ -28,6 +28,19 @@ test_that("convert_mae_to_raw_data works as expected with matrix data", {
   mae2 <- purrr::quietly(runDrugResponseProcessingPipeline)(
     input_df
   )
+
```
[the remark this answers is not in the document]
Comment (bczech) [reply tests/testthat/test-convert_se_to_raw_data.R]: done

File: tests/testthat/test-convert_se_to_raw_data.R (hunk 3 above)
Comment (bczech) [reply tests/testthat/test-convert_se_to_raw_data.R:41]: Single-agent data still exists, only the swap of single agent is reverse and some drugs exists as a Gnumber_2 (to mimic the structure of internal GDS data)

File: R/convert_se_to_raw_data.R [hunk 5]
```diff
@@ -23,11 +23,16 @@ convert_mae_to_raw_data <- function(mae) {
   # Remove duplicates shared between assays to keep only original single-agent
   common_records <- Reduce(intersect, lapply(data, "[[", "record_id"))
   sa_name <- gDRutils::get_experiment_groups("single-agent")[["single-agent"]]
-  data[setdiff(names(data), sa_name)] <- 
-    lapply(data[setdiff(names(data), sa_name)], function(x) {
-      x[!record_id %in% common_records]
-    })
-  
+  combo_name <- gDRutils::get_experiment_groups("combination")
+  if (length(names(data)) > 1 && max(table(unique(data[[combo_name]])[record_id %in% common_records]$record_id)) == 1) {
```
[the remark this answers is not in the document]
Comment (bczech) [reply R/convert_se_to_raw_data.R:27]: done

**Pull request merged**